packages/sakuli-legacy/src/context/common/button-registry.ts

```diff
@@ -0,0 +1,40 @@
+import { Key } from "./key.class";
+import { MouseButton } from "./button.class";
+
+export interface ButtonRegistry {
+  keyboard: Key[];
+  mouse: MouseButton[];
```

Comment:
Do we really have to store each pressed button / key?
Wouldn't it be sufficient to use e.g. a `Map<Key, number>`?

This would still allow us to track the amount of activations, but inserting / looking up / deleting entries would be a typical hash table use-case for me.
Since we're only resetting button state, a single entry should™ be sufficient instead of carrying all possible instances.

Reply:
Using a hash map for inserting/deleting would be an improvement performance wise and the external representation of the button registry would not be affected by this.

Reply:
```quote
Since we're only resetting button state, a single entry should™ be sufficient [...]
```
Therefore I used a Set instead of a hash map.